PrimeAssemblyScript/src/sieve.ts

```diff
@@ -0,0 +1,69 @@
+export class PrimeSieve {
+    sieveSize: i32 = 0;
+    private Bits: StaticArray<bool>;
+    private static primeCounts: Map<i32, i32> = new Map<i32, i32>().set(10, 4)
+        .set(100, 25)
+        .set(1000, 168)
+        .set(10000, 1229)
+        .set(100000, 9592)
+        .set(1000000, 78498)
+        .set(10000000, 664579)
+        .set(100000000, 5761455);
+    public constructor(size: u32) {
+        this.sieveSize = size;
+        // StaticArray did look a bit faster than UInt8Array in testing
+        this.Bits = new StaticArray<bool>(size);
+
+    }
+    validateResults(): bool {
+        if (PrimeSieve.primeCounts[this.sieveSize])
+            return PrimeSieve.primeCounts[this.sieveSize] == this.countPrimes();
+        return false;
+    }
+
+    runSieve(): void {
+        let factor: i32 = 3;
+        let q: i32 = Math.sqrt(this.sieveSize) as i32;
+
+        while (factor <= q) {
+            for (let num: i32 = factor; num < this.sieveSize; num += 2) {
+                if (!this.Bits[num]) {
+                    factor = num;
+                    break;
+                }
+            }
+            for (let num: i32 = factor * factor; num < this.sieveSize; num += factor * 2)
```

Comment:
```suggestion
            for (let num: i32 = factor * factor; num < size; num += factor * 2)
```
depends on previous suggestion